cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/RouteConfigHttpHandler.java

```diff
@@ -78,7 +78,14 @@ public void storeRouteConfig(HttpRequest request, HttpResponder responder,
                                @PathParam("service-id") String serviceId) throws Exception {
     ProgramId programId = Ids.namespace(namespaceId).app(appId).service(serviceId);
     Map<String, Integer> routes = parseBody(request, ROUTE_CONFIG_TYPE);
-    routeStore.store(programId, new RouteConfig(routes));
+    RouteConfig routeConfig;
+    try {
+      routeConfig = new RouteConfig(routes);
```

Comment:
Is it though? It seems like a generic exception that anyone could throw - providing HTTP response status is just additional capability? Service classes (that HTTP handlers delegate to) use these exceptions - technically these classes could be used anywhere. I thought there were jiras to even have the ``Store`` interface throw these exceptions.


Don't feel too strongly about this though - just a nit - code could be cleaner that way.